adtrust: print DNS records for external DNS case after role is enabled

abbra opened the pull request:
We cannot gather information about required DNS records before "ADTrust
Controller" role is enabled on this server. As result, we need to call
the step to add DNS records after the role was enabled.

Fixes: https://pagure.io/freeipa/issue/8192
Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>

tiran commented on ipaserver/install/adtrust.py:
```diff
@@ -435,6 +435,9 @@ def install(standalone, options, fstore, api):
         add_new_adtrust_agents(api, options)
 
+    # Enable configured services and update DNS SRV records
+    service.sync_services_state(api.env.host)
```
The ``adtrust.install()`` is also called from the ``ipa-server-install`` script. This change may enable the server a bit too early if the installer is called with ``--setup-adtrust``.

abbra replied:
.. and also from `ipa-replica-install` too. I'll check these paths.

flo-renaud commented:
Hi @abbra 
overall the change looks good but when ipa-adtrust-install is run on a system with DNS enabled, the DNS records should get automatically added and it's not happening any more.

abbra commented:
@flo-renaud thanks for the review. I see that I removed one unconditional DNS record updates. I replaced it now with a call to update it only if no DNS help needs to be printed. This simulates the behavior of the original code.

flo-renaud commented:
Hi @abbra 
thanks for the update, LGTM. Tested the following:
-[x] ipa-server-install without DNS: prints all the missing records in a file
-[x] ipa-adtrust-install without DNS: prints the missing trust-related records on stdout
-[x] ipa-replica-install with AD trust without DNS: prints the missing trust-related records on stdout
-[x] ipa-server-install with AD trust without DNS: prints the missing trust-related records on stdout + all records in a file
When DNS is setup, there is no message about missing records, as expected.

abbra commented:
master:

* b3dbb36867ebf52483834ceb247050a24cf74d7c adtrust: print DNS records for external DNS case after role is enabled
